src/nccl_ofi_interface_nvidia.c

```diff
@@ -32,8 +32,36 @@ static ncclResult_t getProperties_v8(int dev_id, ncclNetProperties_v8_t* props)
 	 * support this model (since NCCL maintains a per-domain registration
 	 * cache which requires (domain-)global registrations.
 	 */
-	if (ofi_properties.mr_scope == NCCL_OFI_MR_SCOPE_DOMAIN)
-		props->regIsGlobal = 1;
+	if (ofi_properties.mr_scope == NCCL_OFI_MR_SCOPE_DOMAIN) {
+		//props->regIsGlobal = 1;
+		/**
+		 * TODO:
+		 * When net-plugin returns regIsGlobal=1 to NCCL (As part of
+		 * net-plugin getProperties() API), it signals to NCCL that
+		 * registered MRs are global, in the sense that they can be
+		 * used by all communicators. In addition, it also signals to
+		 * NCCL that the net-plugin have a fast MR cache such that
+		 * calling regMr() on same buffer (address and size), will
+		 * quickly return a previously globally registered MR on same
+		 * buffer.
+		 *
+		 * When user registers a buffer with NCCL by using
+		 * ncclCommRegister() API, if net-plugin supports
+		 * regIsGlobal=1, NCCL will register the buffer globally once
+		 * (On each net device) with regMr() API. When the net
+		 * proxy-thread starts to execute a communication task on a
+		 * previously registered user buffer, it will call the
+		 * net-plugin regMr() to quickly fetch the previously globally
+		 * registered MR from the plugin managed MR cache.
+		 *
+		 * Even though aws-ofi-nccl registers MRs globally (As MRs
```

Comment:
This comment is not always true.  It is only true in this particular condition (mr_scope == NCCL_OFI_MR_SCOPE_DOMAIN), but your comment makes some pretty blanket statements.

This code is correct, but the obvious fix when we add the cache will make this code not correct.  It's true that both protocols do not support `regIsGlobal`, but this comment and code implies that we can flip this to 1 and be done.  But either this code or the code that sets `ofi_properties.mr_scope` needs to account for the sendrecv protocol's send case not reporting truncation properly.

Reply:
Updated comment